adapters/sidekiq/Appraisals

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+appraise 'sidekiq-6.0' do
+  gem 'sidekiq', '~> 6.0'
```

Comment:
I'm not familiar with the best practices around using Appraisal.  What releases should be included for appraisal?

CI is also [failing](https://circleci.com/gh/open-telemetry/opentelemetry-ruby/1717?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link) on the `6.0` release of sidekiq when testing against ruby2.4 which is to be expected.

>Ruby <2.5 is no longer supported.

https://github.com/mperham/sidekiq/blob/master/6.0-Upgrade.md#upgrade

Reply:
I'm not sure that we have established any hard rules about what versions we should test using appraisal. At a minimum, I'd suggest that we test against the major versions in the supported range, and consider testing against the latest minor versions. In general, I think we just want reasonable assurance the adapter works for the versions we expect it to.

Reply:
[Ruby 2.4 support has ended](https://www.ruby-lang.org/en/news/2020/04/05/support-of-ruby-2-4-has-ended/). In #10 we decided not to support versions not supported by Ruby Core. We should bump the minimum to 2.5 everywhere, which will fix this problem too. I’ll do that in another PR and then this can be rebased.

Reply:
If we run into this again, which is probably likely, we can remove the generated gemfiles from source control, and generate them before each run. That would allow us to put conditionals in the Appraisals file based on version. For example:

```ruby
if RUBY_VERSION >= '2.5.0'
  appraise 'sidekiq-6.0' do
    gem 'sidekiq', '~> 6.0'
  end
end
```

We'd just have to add `bundle exec rake appraisal generate` to the circle ci script.

Reply:
https://github.com/open-telemetry/opentelemetry-ruby/pull/226 removes Ruby 2.4 support.